src/main/java/org/opentripplanner/transit/raptor/rangeraptor/transit/TransitCalculator.java

```diff
@@ -157,11 +157,11 @@
     TripScheduleSearch<T> createExactTripSearch(RaptorTimeTable<T> timeTable);
 
     /**
-     * Return a guaranteed transfer provider for the given pattern. When searching forward the
+     * Return a transfer provider for the given pattern. When searching forward the
      * given {@code target} is the TO pattern/stop, while when searching in reverse the given
      * target is the FROM pattern/stop.
      */
-    RaptorGuaranteedTransferProvider<T> guaranteedTransfers(RaptorRoute<T> route);
+    RaptorTransferConstraintSearch<T> transferConstraintsSearch(RaptorRoute<T> route);
```

Comment:
Is it `TransferConstraintsSearch` or `ConstraintTransfersSearch`?

Reply:
This should probably have a better name. The interface is used by Raptor to find "special" trip boardings. That, is boarding as a result of a constrained transfer. So, maybe `RaptorConstrainedTransferSearch` is a better name. It do not return transfers, it returns BoadingEvents with a transfer constraint attached.

Reply:
What about `RaptorConstrainedTripScheduleBoardingSearch` - the name is long but I am find with long names.